corehq/util/context_processors.py

```diff
@@ -99,9 +99,11 @@ def websockets_override(request):
 def enterprise_mode(request):
     return {
         'enterprise_mode': settings.ENTERPRISE_MODE,
-        'is_saas_environment': settings.IS_SAAS_ENVIRONMENT
+        'is_saas_environment': settings.IS_SAAS_ENVIRONMENT,
+        'restrict_domain_creation': settings.RESTRICT_DOMAIN_CREATION,
```

Comment:
you could do the superuser check here so templates only need to check the one thing.